src/runtime-rs/crates/runtimes/common/src/types/mod.rs

```diff
@@ -0,0 +1,209 @@
+    CloseIo(ProcessID),
+    DeleteProcess(ProcessID),
+    Exec(ExecProcessRequest),
+    KillProcess(KillRequest),
+    Wait(ProcessID),
+    StartProcess(ProcessID),
+    StateProcess(ProcessID),
+    Shutdown(ShutdownRequest),
+    Pause(String),
+    Resume(String),
+    ResizePty(ResizePtyRequest),
+    Stats(String),
+    Update(UpdateRequest),
+    Pid,
+    Connect(String),
+}
+
+impl fmt::Display for Request {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum Response {
+    CreateContainer(ShimPid),
+    CloseIo,
+    DeleteProcess(ProcessStateInfo),
+    Exec,
+    KillProcess,
+    Wait(ProcessExitStatus),
+    StartProcess(ShimPid),
+    StateProcess(ProcessStateInfo),
+    Shutdown,
+    Pause,
+    Resume,
+    ResizePty,
+    StatsResponse(StatsInfo),
+    Update,
+    Pid(ShimPid),
+    Connect(ShimPid),
+}
+
+impl std::fmt::Display for Response {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ProcessType {
+    Container,
+    Exec,
+}
+
+#[derive(Clone, Debug)]
+pub struct ProcessID {
+    pub cid: String,
+    pub eid: String,
+    pub process_type: ProcessType,
+}
+
+impl fmt::Display for ProcessID {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+impl ProcessID {
+    pub fn new(cid: &str, eid: &str) -> Self {
+        let (eid, process_type) = if eid.is_empty() || cid == eid {
+            ("".to_string(), ProcessType::Container)
+        } else {
+            (eid.to_string(), ProcessType::Exec)
+        };
+        Self {
+            cid: cid.to_string(),
+            eid,
+            process_type,
+        }
+    }
+}
+#[derive(Debug, Clone)]
+pub struct ContainerConfig {
+    pub cid: String,
+    pub bundle: String,
+    pub rootfs: Vec<Mount>,
+    pub terminal: bool,
+    pub stdin: Option<String>,
+    pub stdout: Option<String>,
+    pub stderr: Option<String>,
+}
+
+#[derive(Debug, Clone)]
+pub struct ShimPid {
+    pub pid: u32,
+}
+
+impl ShimPid {
+    pub fn new(pid: u32) -> Self {
+        Self { pid }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct KillRequest {
+    pub process_id: ProcessID,
+    pub signal: u32,
+    pub all: bool,
+}
+
+#[derive(Debug, Clone)]
+pub struct ShutdownRequest {
+    pub cid: String,
+    pub is_now: bool,
+}
+
+#[derive(Debug, Clone)]
+pub struct ResizePtyRequest {
+    pub process_id: ProcessID,
+    pub width: u32,
+    pub height: u32,
+}
+
+#[derive(Debug, Clone)]
+pub struct ExecProcessRequest {
+    pub process_id: ProcessID,
+    pub terminal: bool,
+    pub stdin: Option<String>,
+    pub stdout: Option<String>,
+    pub stderr: Option<String>,
+    pub spec_type_url: String,
+    pub spec_value: Vec<u8>,
+}
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum ProcessStatus {
```

Comment:
- The order of these enums (but not their values) matches the containerd task status almost exactly, apart from the fact that there is no `Deleted`.
- What state does the process move to when `Request::DeleteProcess` occurs?
- Does this need a warning on it stating that these values must not be changed?

Reply:
Thanks for your review, I think it's unnecessary to add Process `Delete` Status because the process can't be found after process delete.